api/v1alpha1/databaseengine_types.go

```diff
@@ -89,15 +85,37 @@ type DatabaseEngineStatus struct {
 	PendingOperatorUpgrades []OperatorUpgrade `json:"pendingOperatorUpgrades,omitempty"`
 
 	// OperatorUpgrade contains the status of the operator upgrade.
-	// This is set only if the `everest.percona.com/upgrade-operator-to` annotation is present.
 	OperatorUpgrade *OperatorUpgradeStatus `json:"operatorUpgrade,omitempty"`
 }
 
+// GetNextUpgradeVersion gets the next version of the operator to upgrade to.
+func (s *DatabaseEngineStatus) GetNextUpgradeVersion() string {
+	if len(s.PendingOperatorUpgrades) == 0 {
+		return ""
+	}
+	if len(s.PendingOperatorUpgrades) == 1 {
+		return s.PendingOperatorUpgrades[0].TargetVersion
+	}
+	next := slices.MinFunc(s.PendingOperatorUpgrades, func(a, b OperatorUpgrade) int {
+		v1 := goversion.Must(goversion.NewVersion(a.TargetVersion))
+		v2 := goversion.Must(goversion.NewVersion(b.TargetVersion))
+		return v1.Core().Compare(v2.Core())
+	})
+	return next.TargetVersion
+}
```

Comment:
How should Everest handle the patch releases? If there is a choice between `1.1.0` and `1.1.1` the function returns 1.1.0, does it mean the user suppose to go through each patch upgrade sequentially? It would be nice if Everest would suggest a single upgrade to the latest patch, but not sure if it's feasible. Anyway, let's at least add the the patch examples to the `api/v1alpha1/databaseengine_test.go`

Reply:
Good point, I think we should skip to the latest patch.

> but not sure if it's feasible
Why not?

Reply:
no particular reason in mind, I just wasn't sure if the upgrade is going to work well with all of the components (catalog, VS) but seems like it should.